product_harmonized_system/views/product_template.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+  Copyright 2010-2024 Akretion France (http://www.akretion.com/)
+  @author Alexis de Lattre <alexis.delattre@akretion.com>
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
+<odoo>
+    <!-- product.template form view -->
+    <record id="product_template_form_view" model="ir.ui.view">
+        <field name="name">hs_code.product.template.form</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_form_view" />
+        <field name="arch" type="xml">
+            <group name="group_general" position="inside">
+                <field name="hs_code_id" invisible="type == 'service'" />
+                <field
+                    name="origin_country_id"
+                    invisible="type == 'service'"
+                    options="{'no_create': True}"
+                />
+            </group>
+        </field>
+    </record>
+    <record id="product_template_search_view" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view" />
+        <field name="arch" type="xml">
+            <filter name="categ_id" position="after">
+                <filter
+                    string="H.S. Code"
+                    name="hs_code_groupby"
+                    context="{'group_by': 'hs_code_id'}"
+                />
```

Comment:
```suggestion
                <filter
                    string="H.S. Code"
                    name="hs_code_groupby"
                    context="{'group_by': 'hs_code_id'}"
                />
                <filter
                    string="Country of Origin"
                    name="origin_country_groupby"
                    context="{'group_by': 'origin_country_id'}"
                />
```
I guess we can also add the Country of Origin group by, just as an extra improvement.